prql-elixir/native/prql/src/lib.rs

```diff
@@ -113,8 +113,7 @@ pub struct Response {
 #[rustler::nif]
 /// compile a prql query into sql
 pub fn compile(prql_query: &str, options: CompileOptions) -> NifResult<Response> {
-    let result = prql_compiler::compile(prql_query, Some(options.into()));
-    to_result_tuple(result)
+    to_result_tuple(prql_compiler::compile(prql_query, Some(options.into())));
```

Comment:
Need to remove `;`

Reply:
@max-sixty i think build for elixir is failing because of the `;`

Reply:
Should be `to_result_tuple(prql_compiler::compile(prql_query, Some(options.into())))`

Reply:
Sorry!

Reply:
:)

Reply:
I'm pushing a fix.

Because this crate isn't part of the workspace, it doesn't get built with the usual tests, so is liable to fail in this sort of error. But:
- we'll see the error after merge, when `test-elixir.yaml` runs
- we can look into making it part of the workspace, on reviewing the docs I think it might be possible (though maybe not a good idea)

Reply:
Im not sure whether we can make it a part of the workspace as elixir bindings are totally different from other bindings we have. 

Can we add `test-elixir.yaml` for the default test suite to avoid this kind of errors?

Reply:
I'm out of my depth so I won't push on this — I don't even know whether it's overall a good idea.

But this does pass the tests!

```diff
diff --git a/Cargo.toml b/Cargo.toml
index ed29d034..96610687 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -6,6 +6,7 @@ members = [
   "prql-lib",
   "prql-js",
   "prql-python",
+  "prql-elixir/native/prql",
   "book",
 ]
 # Note we don't have a `default-members = ["prql-compiler"]`, since that causes
diff --git a/prql-elixir/native/prql/Cargo.toml b/prql-elixir/native/prql/Cargo.toml
index 0fab6646..a3636472 100644
--- a/prql-elixir/native/prql/Cargo.toml
+++ b/prql-elixir/native/prql/Cargo.toml
@@ -1,10 +1,10 @@
 [package]
 name = "prql"
 description = "Elixir NIF bindings for prql-compiler"
-version = "0.1.0"
 license = "Apache-2.0"
 authors = ["Kasun Vithanage <kasvith.me@gmail.com>"]
 edition = "2018"
+version.workspace = true
 
 [lib]
 name = "prql"
@@ -14,5 +14,3 @@ crate-type = ["cdylib"]
 [dependencies]
 rustler = "0.26.0"
 prql-compiler = {path = "../../../prql-compiler", default-features = false}
-
-[workspace]

```

Reply:
If we add a single OS and a Single OTP version, we can get builds pretty fast(check ubuntu test)

Reply:
> I'm out of my depth so I won't push on this — I don't even know whether it's overall a good idea.
> 
> But this does pass the tests!
> 
> ```diff
> diff --git a/Cargo.toml b/Cargo.toml
> index ed29d034..96610687 100644
> --- a/Cargo.toml
> +++ b/Cargo.toml
> @@ -6,6 +6,7 @@ members = [
>    "prql-lib",
>    "prql-js",
>    "prql-python",
> +  "prql-elixir/native/prql",
>    "book",
>  ]
>  # Note we don't have a `default-members = ["prql-compiler"]`, since that causes
> diff --git a/prql-elixir/native/prql/Cargo.toml b/prql-elixir/native/prql/Cargo.toml
> index 0fab6646..a3636472 100644
> --- a/prql-elixir/native/prql/Cargo.toml
> +++ b/prql-elixir/native/prql/Cargo.toml
> @@ -1,10 +1,10 @@
>  [package]
>  name = "prql"
>  description = "Elixir NIF bindings for prql-compiler"
> -version = "0.1.0"
>  license = "Apache-2.0"
>  authors = ["Kasun Vithanage <kasvith.me@gmail.com>"]
>  edition = "2018"
> +version.workspace = true
>  
>  [lib]
>  name = "prql"
> @@ -14,5 +14,3 @@ crate-type = ["cdylib"]
>  [dependencies]
>  rustler = "0.26.0"
>  prql-compiler = {path = "../../../prql-compiler", default-features = false}
> -
> -[workspace]
> ```

Does it pass `mix test` also?

Reply:
It passes `test-elixir`!

Reply:
> If we add a single OS and a Single OTP version, we can get builds pretty fast(check ubuntu test)

Ubuntu comes in at 1:57. So possibly. Let's run it for a while on `main` and then decide.

Reply:
awesome, i will also double check the NIF build with workspace settings 

Then we can have greenlight

Reply:
@max-sixty green 🟢 from me. I tested locally with workspace settings and everything works fine.
We can use this 

Thanks for the solution